internal/resources/google/network_egress_internal.go

```diff
@@ -0,0 +1,306 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+type EgressResourceType int
+
+const (
+	StorageBucketEgress EgressResourceType = iota
+	ContainerRegistryEgress
+	ComputeVPNGateway
+	ComputeExternalVPNGateway
+)
+
+type NetworkEgressUsage struct {
+	SameContinent *float64 `infracost_usage:"same_continent"`
+	SameRegion    *float64 `infracost_usage:"same_region"`
+	USOrCanada    *float64 `infracost_usage:"us_or_canada"`
+	Europe        *float64 `infracost_usage:"europe"`
+	Asia          *float64 `infracost_usage:"asia"`
+	SouthAmerica  *float64 `infracost_usage:"south_america"`
+	Oceania       *float64 `infracost_usage:"oceania"`
+	Worldwide     *float64 `infracost_usage:"worldwide"`
+	China         *float64 `infracost_usage:"china"`
+	Australia     *float64 `infracost_usage:"australia"`
+}
+
+var NetworkEgressUsageSchema = []*schema.UsageItem{
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "same_continent"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "same_region"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "us_or_canada"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "europe"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "asia"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "south_america"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "oceania"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "worldwide"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "china"},
+	{ValueType: schema.Float64, DefaultValue: 0, Key: "australia"},
+}
+
+type egressRegionData struct {
+	gRegion             string
+	apiDescription      string
+	apiDescriptionRegex string
+	usageKey            string
+	fixedRegion         string
```

Comment:
Although this isn't related to these changes, it might be good to document what `gRegion` and `fixedRegion` are. It's not immediately clear to me looking through the code.